Automattic/wp-calypso
# WooCommerce setup improvements

#### Changes proposed in this Pull Request

This PR does the following:
- Fix unintended timeout occurring between engine intervals by destroying the timeout after finding the plugin present.
- Increasing the plugin installation and activation time limit to 60 seconds.
- Add a new track `calypso_woocommerce_setup_contact_support` to better understand user behaviour after experiencing setup timeout or failure.

#### Testing instructions

1. Run calypso in your development environment.
2. Manually induce stalling:
    - Open up `client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx`
    - Search for `this.props.installAndActivatePlugin( site.ID, plugin );`.
    - Comment out the line and wait for calypso to rebuild.
4. Create a new business site.
5. Install `Hello Dolly` plugin to turn it into Atomic site.
6. Navigate to `/woocommerce-installation/{your new site}`.
7. Click on `Set up my store!`.
8. Wait for approximately 1 minute, and observe "contact support" screen appearing.
9. Open up developer console, type in `localStorage.setItem( 'debug', 'calypso:analytics*' )` to debug tracks.
10. Click on "Get in touch"
11. Observe the track `calypso_woocommerce_setup_contact_support` is recorded in console containing `reason: 'timeout'`.

Related to #50833
